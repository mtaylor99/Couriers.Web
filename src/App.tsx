/* eslint-disable no-nested-ternary */
import { useEffect, useState } from "react";
import { MuiPickersUtilsProvider } from "@material-ui/pickers";
import { ToastContainer } from "react-toastify";
import DateFnsUtils from "@date-io/date-fns";
import { Grid } from "@material-ui/core";
import IndexRoutes from "./pages/IndexRoutes";
import { getJobStatuses } from "./api/thunks/jobs";
import getScheduleStatuses from "./api/thunks/schedules";
import getDrivers from "./api/thunks/drivers";
import { useAppDispatch } from "./state";
import { SpinnerWithLabel } from "./common/components/spinners";
import { ErrorMessage } from "./common/components/errorMessage";
import DefaultTopBar from "./common/components/topBars";
import { Footer } from "./common/components/footer";
import "react-toastify/dist/ReactToastify.css";
// import TokenRefresher from "./feature/authentication/components/tokenRefresher/TokenRefresher";

const App = () => {
  const [loading, setLoading] = useState(true);
  const [error, setError] = useState(false);
  const dispatch = useAppDispatch();

  useEffect(() => {
    Promise.all([
      dispatch(getJobStatuses() as any),
      dispatch(getScheduleStatuses() as any),
      dispatch(getDrivers() as any),
    ])
      .catch(() => {
        setError(true);
      })
      .finally(() => setLoading(false));
  }, [dispatch]);

  return (
    <>
      <MuiPickersUtilsProvider utils={DateFnsUtils}>
        {/* <TokenRefresher /> */}
        <ToastContainer />
        {loading ? (
          <SpinnerWithLabel centerScreen label="Loading" />
        ) : error ? (
          <ErrorMessage
            centerScreen
            reason="A error occcured while loading the application."
          />
        ) : (
          <Grid container direction="column">
            <Grid item xs={12}>
              <DefaultTopBar />
            </Grid>
            <Grid container item>
              <Grid item xs={12}>
                <IndexRoutes />
              </Grid>
            </Grid>
            <Grid item xs={12}>
              <Footer />
            </Grid>
          </Grid>
        )}
      </MuiPickersUtilsProvider>
    </>
  );
};

export default App;
