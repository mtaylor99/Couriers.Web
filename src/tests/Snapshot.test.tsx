import { waitFor } from "@testing-library/react";
import { render } from "./testUtils";
import mockFetch from "./mockFetch";
import App from "../App";

describe("With React Testing Library", () => {
  it("Snapshot for App component", async () => {
    const { container, getByText } = render(<App />);

    await waitFor(() => getByText("Dashboard Page"));

    expect(container).toMatchSnapshot();
  });
});

beforeAll(() => jest.spyOn(window, "fetch"));
beforeEach(() => (window.fetch as jest.Mock).mockImplementation(mockFetch));
