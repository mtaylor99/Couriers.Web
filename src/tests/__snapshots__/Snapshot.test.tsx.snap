// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`With React Testing Library Snapshot for App component 1`] = `
<div>
  <div
    class="Toastify"
  />
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column"
  >
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
    >
      <header
        class="MuiPaper-root MuiAppBar-root MuiAppBar-positionSticky MuiAppBar-colorPrimary WithStyles(ForwardRef(AppBar))-root-3 WithStyles(ForwardRef(AppBar))-root-4 MuiPaper-elevation0"
      >
        <div
          class="div-root-5 div-root-6"
        >
          <a
            href="/"
          >
            <img
              alt="TNHG logo"
              src="logo.svg"
            />
          </a>
        </div>
        <div
          class="div-root-7 div-root-8"
        >
          <a
            href="/"
          >
            Dashboard
          </a>
        </div>
        <div
          class="div-root-7 div-root-9"
        >
          <a
            href="/jobs"
          >
            Jobs
          </a>
        </div>
        <div
          class="div-root-7 div-root-10"
        >
          <a
            class="MuiTypography-root MuiLink-root MuiLink-underlineHover WithStyles(ForwardRef(Link))-root-11 WithStyles(ForwardRef(Link))-root-12 MuiTypography-colorPrimary"
            href="need-help"
            target="_blank"
          >
            Need Help?
          </a>
        </div>
      </header>
    </div>
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-item"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
      >
        <div>
          Dashboard Page
        </div>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
    >
      <div
        class="MuiContainer-root WithStyles(ForwardRef(Container))-root-13 WithStyles(ForwardRef(Container))-root-14 MuiContainer-maxWidthLg"
      >
        <div
          class="div-root-15 div-root-16"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained WithStyles(ForwardRef(Button))-root-17 WithStyles(ForwardRef(Button))-root-18 MuiButton-disableElevation"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              Terms and Conditions
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
          <span
            class="span-root-19 span-root-20"
          >
            |
          </span>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained WithStyles(ForwardRef(Button))-root-17 WithStyles(ForwardRef(Button))-root-21 MuiButton-disableElevation"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              Privacy
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
          <span
            class="span-root-19 span-root-22"
          >
            |
          </span>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained WithStyles(ForwardRef(Button))-root-17 WithStyles(ForwardRef(Button))-root-23 MuiButton-disableElevation"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              Cookies
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
        <div
          class="div-root-24 div-root-25"
        >
          Â© Copyright 2021 | The New Home Group
        </div>
      </div>
    </div>
  </div>
</div>
`;
